src/store/middlewares/notificationMiddleware.tsx

```diff
@@ -0,0 +1,104 @@
+import { Middleware } from '@reduxjs/toolkit';
+import { Logger } from 'edumeet-common';
+import { chatActions } from '../slices/chatSlice';
+import { filesharingActions } from '../slices/filesharingSlice';
+import { lobbyPeersActions } from '../slices/lobbyPeersSlice';
+import { peersActions } from '../slices/peersSlice';
+import { signalingActions } from '../slices/signalingSlice';
+import { MiddlewareOptions, RootState } from '../store';
+
+interface SoundAlert {
+	[type: string]: {
+		audio: HTMLAudioElement;
+		debounce: number;
+		last?: number;
+	};
+}
+
+const logger = new Logger('NotificationMiddleware');
+
+const createNotificationMiddleware = ({
+	config
+}: MiddlewareOptions): Middleware => {
+	logger.debug('createNotificationMiddleware()');
+
+	const soundAlerts: SoundAlert = {
+		'default': {
+			audio: new Audio('/sounds/notify.mp3'),
+			debounce: 0
+		}
+	};
+
+	const playNotificationSounds = async (type: string) => {
+		const soundAlert = soundAlerts[type] ?? soundAlerts['default'];
+
+		const now = Date.now();
+
+		if (soundAlert?.last && (now - soundAlert.last) < soundAlert.debounce)
+			return;
+
+		soundAlert.last = now;
+
+		await soundAlert.audio.play().catch((error) => {
+			logger.error('soundAlert.play() [error:"%o"]', error);
+		});
+	};
+
+	const middleware: Middleware = ({
+		getState
+	}: {
+		getState: () => RootState
+	}) =>
+		(next) => async (action) => {
+			if (signalingActions.connect.match(action) && config.notificationSounds) {
```

Comment:
Hmm...I think this belongs inside the factory function above, no need to have this inside the middleware. That way it is loaded as soon as the store is initialized.

Reply:
Fixed